cai_causal_graph/time_series_causal_graph.py

```diff
@@ -126,6 +198,8 @@ def __eq__(self, other: object) -> bool:
         # now check the metadata timedelta in the nodes
         for node in self.get_nodes():
             other_node = other.get_node(node.identifier)
+            assert isinstance(other_node, TimeSeriesNode)
```

Comment:
Would add `# For linting` after this so it doesn't confuse things

Reply:
+1